src/profiler.cpp

```diff
@@ -810,6 +810,11 @@ Error Profiler::installTraps(const char* begin, const char* end, bool nostop) {
         return Error("End address not found");
     }
 
+    // begin & end shouldn't have the same address, as that would result in an infinite loop
+    if (begin_addr && begin_addr == end_addr) {
+        return Error("begin & end should be 2 different symbols");
```

Comment:
I propose `begin and end symbols should not resolve to the same address`. They can be different symbols with the same address.